libpod/runtime_ctr.go

```diff
@@ -291,6 +296,13 @@ func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (c *Contai
 		ctr.config.Mounts = append(ctr.config.Mounts, ctr.config.ShmDir)
 	}
 
+	// Lock all named volumes we are adding ourself to, to ensure we can't
+	// use a volume being removed.
+	for _, namedVol := range ctrNamedVolumes {
+		namedVol.lock.Lock()
```

Comment:
That's the golang-pattern of death :skull: 

We need to lock/unlock on `ctrNamedVolumes[i]` since the loop-variable is just a pointer such that the deferred references always point to the last item of the loop. So if the `ctrNamedVolumes` slice has two items, we will unlock the last item twice (due to the defer) while the first one remains unlocked.